apps/condo/domains/property/gql.js

```diff
@@ -5,13 +5,61 @@
  */
 
 const { generateGqlQueries } = require('@condo/domains/common/utils/codegeneration/generate.gql')
+const { ADDRESS_META_SUBFIELDS_QUERY_LIST } = require('@condo/domains/common/utils/addressApi/addressMetaGql')
 const { gql } = require('graphql-tag')
 
 const COMMON_FIELDS = 'id dv sender { dv fingerprint } v deletedAt organization { id name} newId createdBy { id name } updatedBy { id name } createdAt updatedAt'
-
-const PROPERTY_FIELDS = `{ name address addressMeta type ticketsInWork ticketsClosed unitsCount map ${COMMON_FIELDS} }`
+const PROPERTY_MAP_JSON_FIELDS = 'dv type sections { id type index name preview floors { id type index name units { id type name label preview } } }'
+const PROPERTY_FIELDS = `{ name address addressMeta { ${ADDRESS_META_SUBFIELDS_QUERY_LIST} } type ticketsInWork ticketsClosed unitsCount map { ${PROPERTY_MAP_JSON_FIELDS} } ${COMMON_FIELDS} }`
 const Property = generateGqlQueries('Property', PROPERTY_FIELDS)
 
+const PROPERTY_MAP_GRAPHQL_TYPES = `
+    scalar building
+    scalar section
+    scalar floor
+    scalar unit
+    scalar village
+
+    enum BuildingMapEntity {
+        building
+        section
+        floor
+        unit
+        village
+    }
+
+    type BuildingUnit {
+        id: String!
+        type: unit!
+        name: String
+        label: String!
+        preview: Boolean
+    }
+
+    type BuildingFloor {
+        id: String!
+        type: floor!
```

Comment:
BuildingMapEntityType

Reply:
It must have only one value — "floor", this is the only way, I've found, to ensure this exact value.
When I will use enum type, any value from enum can slip through GraphQL input validation.